File: naive_approach.go
package main

import "fmt"

func stringMatching(str string, pattern string) bool {
	for i := 0; i < len(str); i++ {
		j := 0
		for j < len(pattern) {
			if str[i+j] != pattern[j] {
				break
			}
			j += 1
		}
		if j == len(pattern) {
			return true
		}
	}
	return false
}

func main() {
	str := "kmppkmpkmpkmdkmpkmpk"
	pattern := "kmpkmdkmpkmpk"
	fmt.Println("does match:", stringMatching(str, pattern))
}

/*
output:
does match: true
*/

File: naive_approach.py
def stringMatching(string, pattern):
    for i in range(len(string)):
        j = 0
        while j < len(pattern):
            if string[i+j] != pattern[j]:
                break
            j += 1
        if j == len(pattern):
            return True
    return False


if __name__ == "__main__":
    string = "kmppkmpkmpkmdkmpkmpk"
    pattern = "kmpkmdkmpkmpk"
    print("does match:", stringMatching(string, pattern))

"""
output:
('does match:', True)
"""